### PR TITLE
[docs] refactor docs to cleanup repo look

### LINKED ISSUE #586: [docs] rework documentation for modules in GW
Move the documentation per module into respective module folders. i.e. add a new `README.md` file under each module such as `cudapoa` or `cudaextender` and add more detailed documentation regarding algorithm, features, limitations, etc in there.

### PR DESCRIPTION
**Pull request by tijyojwad**
Closes #586 

Refactors documentation into module specific READMEs.

### CODE REVIEW

File: cudaaligner/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# cudaaligner
+
+The `cudaaligner` package provides GPU-accelerated global alignment. Features include:
```
Comment (atadkase) [inline cudaaligner/README.md:3]: Features include seems incomplete.

File: cudapoa/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+## Tool
+
+A command line tool for generating consensus and MSA from a list of `fasta`/`fastq` files. The tool
+is built on top of `libcudapoa.so` and showcases optimization strategies for writing high performance
+applications with `libcudapoa.so`.
+
+## Library
+Built as `libcudapoa.[so|a]`
+
+* Generation of consensus sequences
```
Comment (atadkase) [inline cudapoa/README.md:15]: I think these points are supposed to be above the "Tool" section.

**Review by atadkase — CHANGES_REQUESTED** [review]
Some rearrangements required in two readmes. Others look good.

**Comment by atadkase** [comment]
Also it's awesome that you were able to get the Readmes to link together!

File: cudapoa/README.md (hunk 2 above)
Comment (tijyojwad) [reply cudapoa/README.md:9]: They're meant to be part of the library section, but I reorganized and moved the library component to the top.

File: cudaaligner/README.md (hunk 1 above)
Comment (tijyojwad) [reply cudaaligner/README.md]: Yes this was misleading. The `:` was for the `Library` and `Samples` section. But removed it to make it clearer.

File: cudapoa/README.md (hunk 2 above)
Comment (atadkase) [reply cudapoa/README.md:9]: I was confused by the features include part in the cudapoa description too. Can you please remove that opening?

**Review by atadkase — APPROVED** [review]
Looks good except for the small cudapoa nit. It's okay to resolve the conversation and merge it if it's addressed.

File: cudapoa/README.md (hunk 2 above)
Comment (tijyojwad) [reply cudapoa/README.md:9]: totally missed that. Done

**Pull request merged**